Fix Bug 7223: Net::DNS 1.01 breaks DnsResolver

This bug is already fixed for version 3.4.2 and later.

See: https://bz.apache.org/SpamAssassin/show_bug.cgi?id=7223

--- a/lib/Mail/SpamAssassin/DnsResolver.pm
+++ b/lib/Mail/SpamAssassin/DnsResolver.pm
@@ -592,6 +592,9 @@
   };
 
   if ($packet) {
+    # RD flag needs to be set explicitly since Net::DNS 1.01, Bug 7223	
+    $packet->header->rd(1);
+
   # my $udp_payload_size = $self->{res}->udppacketsize;
     my $udp_payload_size = $self->{conf}->{dns_options}->{edns};
     if ($udp_payload_size && $udp_payload_size > 512) {
